test/e2e/gameserver_test.go

```diff
@@ -522,17 +525,40 @@ func TestGameServerWithPortsMappedToMultipleContainers(t *testing.T) {
 	defer framework.AgonesClient.AgonesV1().GameServers(defaultNs).Delete(readyGs.ObjectMeta.Name, nil) // nolint: errcheck
 	assert.Equal(t, readyGs.Status.State, agonesv1.GameServerStateReady)
 
-	firstContainerReply, err := e2eframework.SendGameServerUDPToPort(readyGs, "gameport", "Ping 1")
-	if err != nil {
-		t.Fatalf("Could not message GameServer: %v", err)
-	}
-	assert.Contains(t, firstContainerReply, "Ping 1")
+	interval := 2 * time.Second
+	timeOut := 60 * time.Second
 
-	secondContainerReply, err := e2eframework.SendGameServerUDPToPort(readyGs, "second-gameport", "Ping 2")
-	if err != nil {
-		t.Fatalf("Could not message GameServer: %v", err)
-	}
-	assert.Contains(t, secondContainerReply, "Ping 2")
+	pollCh1 := make(chan error)
+	pollCh2 := make(chan error)
+
+	go func() {
```

Comment:
Is there a reason this couldn't be 2 polling statements, one after another?

Then it would make this test much simpler, and remove the need for channels. I don't think we have to test for both UDP endpoints at the same time.

WDYT?

Reply:
The main idea why I've introduced 2 goroutines is that in a worst-case scenario if we ping ports one by one, this **single** test will run 2 mins at least - there are 2 timeouts, 60 sec for each ping. According to my tests, there is a little chance of this scenario, but I decided to take it into account. I've tried to make this code as simple as possible. 
Do you think we need to call polling statements one after another anyway?

Reply:
Since failure should be an exception not the regular path, I would have them one after the other, for sake of simplicity of the testing code. This would make it easier to maintain going forward.

One optimisation you could make that would cut things down time in failure scenarios, is that if the first one fails, do a `assert.FailNow()`; to fast fail the test and stop the second one from even starting.

Reply:
Ok, good suggestion!